File: day_10/puzzle_2/solve.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = __importDefault(require("fs"));
const benchmark_1 = require("../../benchmark/benchmark");
(0, benchmark_1.benchmark)(() => {
    const file = fs_1.default.readFileSync("./day_10/input.txt", { encoding: "utf-8" });
    let x = 1;
    let cycleCount = 0;
    const crt = new Array(6).fill("");
    const cycle = () => {
        cycleCount++;
        const row = Math.floor((cycleCount - 1) / 40);
        if (Math.abs(((cycleCount - 1) % 40) - x) <= 1) {
            crt[row] += "#";
        }
        else {
            crt[row] += ".";
        }
    };
    file.split("\n").forEach((instruction) => {
        cycle();
        if (instruction.startsWith("addx")) {
            cycle();
            x += +instruction.split(" ")[1];
        }
    });
    return `\n${crt.join("\n")}`;
}, "day_10_puzzle_1");

File: day_10/puzzle_2/solve.py
import math
import os
import sys

sys.path.append(
    os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
)

from benchmark.benchmark import benchmark


def main():
    with open("./day_10/input.txt", "r") as file:
        x = 1
        cycle_count = 0
        crt = [""] * 6

        def cycle(x, cycle_count, crt):
            cycle_count += 1

            row = math.floor((cycle_count - 1) / 40)

            if abs(((cycle_count - 1) % 40) - x) <= 1:
                crt[row] += "#"
            else:
                crt[row] += "."

            return [cycle_count, crt]

        for instruction in file.readlines():
            cycle_count, crt = cycle(x, cycle_count, crt)

            if instruction.startswith("addx"):
                cycle_count, crt = cycle(x, cycle_count, crt)
                x += int(instruction.split(" ")[1])

        return "\n".join(crt)


if __name__ == "__main__":
    benchmark(main, "day_10_puzzle_2")
